File: LeetCode/Practice Problems/power_of_four.cpp
class Solution {
public:
    bool isPowerOfFour(int n) {
        if(n<=0)
            return false;
        if((n & (n-1))==0)
        {
            int count=0;
            while(n!=1)
            {
                n = n>>1;
                count++;
            }
            if(count%2==0)
                return true;
        }
        return false;
    }
};

File: LeetCode/Daily LeetCoding Challenge/power_of_four.java
class Solution {
    public boolean isPowerOfFour(int n) {
        if(n <= 0)
            return false;    // returning false for negative numbers
        
        if((n & (n - 1)) == 0) {    // checking if the number is power of two or not
            int c = 0;
            while(n != 1) {
                c++;    // counting the number of 0 bits
                n = n >>> 1;
            }
            
            if(c % 2 == 0)    // if no. of 0 bits is even, then n is power of four
                return true;
        }
        
        return false;
    }
}